QSBGameModes/Menu/GameModeMenu.cs

```diff
@@ -66,43 +61,42 @@ private static void UpdateGUI_()
-                    clientText.text = "SPECTATE " + SharedSettings.settingsToShare.GameType;
-                    SetPauseButtonAction(ClientButton, QSBGameModes.LeaveGameMode);
+                    ChangeButtonText(JoinButton, "SPECTATE " + SharedSettings.settingsToShare.GameType);
+                    SetPauseButtonAction(JoinButton, QSBGameModes.LeaveGameMode);
                     Utils.WriteLine($"Setting Leave Button");
                     break;
                 
                 case GameManagement.PlayerManagement.PlayerState.None:
                 case GameManagement.PlayerManagement.PlayerState.Spectating:
-                    clientText.text = "JOIN " + SharedSettings.settingsToShare.GameType;
-                    SetPauseButtonAction(ClientButton, QSBGameModes.JoinGameMode);
+                    ChangeButtonText(JoinButton, "JOIN " + SharedSettings.settingsToShare.GameType);
+                    SetPauseButtonAction(JoinButton, QSBGameModes.JoinGameMode);
                     Utils.WriteLine($"Setting Join Button");
                     break;
             }
         } else {
             if (currentPlayerState is GameManagement.PlayerManagement.PlayerState.None 
                 or GameManagement.PlayerManagement.PlayerState.Spectating){
-                clientText.text =
+                ChangeButtonText(JoinButton, 
                     "READY UP FOR " +
-                    SharedSettings.settingsToShare.GameType; // + SharedSettings.settingsToShare.GameType;
-                SetPauseButtonAction(ClientButton, QSBGameModes.JoinGameMode);
+                    SharedSettings.settingsToShare.GameType); // + SharedSettings.settingsToShare.GameType;
+                SetPauseButtonAction(JoinButton, QSBGameModes.JoinGameMode);
                 Utils.WriteLine($"Setting Ready Up Button");
 
             } else {
-                clientText.text =
+                ChangeButtonText(JoinButton, 
                     "UNREADY UP FOR " +
-                    SharedSettings.settingsToShare.GameType; // + SharedSettings.settingsToShare.GameType;
-                SetPauseButtonAction(ClientButton, QSBGameModes.LeaveGameMode);
+                    SharedSettings.settingsToShare.GameType); // + SharedSettings.settingsToShare.GameType;
+                SetPauseButtonAction(JoinButton, QSBGameModes.LeaveGameMode);
                 Utils.WriteLine($"Setting Unready Up Button");
             }
         }
     }
 
-    public static void SetPauseButtonAction(Button button, UnityAction action){
-        if (button.onClick == null){
-            button.onClick = new Button.ButtonClickedEvent();;
-        }
-        button.onClick.RemoveAllListeners();
-        button.onClick.AddListener(action);
-        button.onClick.AddListener(UpdateGUI);
+    public static void SetPauseButtonAction(SubmitAction button, SubmitAction.SubmitActionEvent action){
+        button.OnSubmitAction += action;
```

Comment:
Before it would clear/remove all the old events so I think you'd probably have to do that somehow